src/iexamine.cpp

```diff
@@ -3465,3 +3389,47 @@ iexamine_function iexamine_function_from_string(std::string const &function_name
     debugmsg("Could not find an iexamine function matching '%s'!", function_name.c_str());
     return &iexamine::none;
 }
+
+hack_result iexamine::hack_attempt(player *p) {
+    if (p->has_trait("ILLITERATE")) {
+        return HACK_UNABLE;
+    }
+    bool using_electrohack = (p->has_amount("electrohack", 1) &&
+                              query_yn(_("Use electrohack?")));
+    bool using_fingerhack = (!using_electrohack && p->has_bionic("bio_fingerhack") &&
+                             p->power_level > 0 &&
+                             query_yn(_("Use fingerhack?")));
+
+    if ( using_electrohack || using_fingerhack ) {
+        p->moves -= 500;
+        p->practice( skill_computer, 20);
+        int success = rng(p->skillLevel( skill_computer ) / 4 - 2, p->skillLevel( skill_computer ) * 2);
+        success += rng(-3, 3);
+        if (using_fingerhack) {
+            success++;
+        }
+        if (p->int_cur < 8) {
+            success -= rng(0, int((8 - p->int_cur) / 2));
+        } else if (p->int_cur > 8) {
+            success += rng(0, int((p->int_cur - 8) / 2));
+        }
```

Comment:
This may be old code, but it seems that both cases (`int < 8` and `int > 8`) contain the same logic.

Why are there separate branches for that? They should be reduced to a single statement.